std-bits/base/src/main/java/org/enso/base/arrays/LongArrayList.java

```diff
@@ -0,0 +1,44 @@
+package org.enso.base.arrays;
+
+import java.util.Arrays;
+
+/** A helper to efficiently build an array of unboxed longs of arbitrary length. */
+public class LongArrayList {
+  private long[] backingStorage;
+  private int lastIndex = -1;
+
+  public LongArrayList() {
+    backingStorage = new long[32];
+  }
+
+  // ** Gets the number of elements in the list. */
+  public int getSize() {
+    return lastIndex + 1;
+  }
+
+  // ** Gets an element from the list. */
+  public long get(int index) {
+    if (index > lastIndex) {
+      throw new IndexOutOfBoundsException(index);
+    }
+    return backingStorage[index];
+  }
+
+  // ** Gets an element from the list. */
+  public long getOrLast(int index) {
+    return backingStorage[Math.min(index, lastIndex)];
+  }
+
+  // ** Adds an element to the list. */
+  public synchronized void add(long x) {
```

Comment:
Why is this `synchronized` when no other method is? The other methods are also accessing `lastIndex` and `backingStore` - they may see inconsistent values.

Reply:
Will remove the synchronized - thought would be enough to synchronize here as was meant to only change the index once the new value was written (admittedly a bug in the code had that the wrong way round).